tests/src/test/scala/ha/ShootComponentsTests.scala

```diff
@@ -63,18 +60,61 @@ class ShootComponentsTests extends FlatSpec with Matchers with WskTestHelpers wi
   val controller0DockerHost = WhiskProperties.getBaseControllerHost() + ":" + WhiskProperties.getProperty(
     WhiskConfig.dockerPort)
 
-  def restartComponent(host: String, component: String) = {
+  val couchDB0DockerHost = WhiskProperties.getBaseDBHost() + ":" + WhiskProperties.getProperty(
+    WhiskConfig.dockerPort)
+
+  val dbProtocol = WhiskProperties.getProperty(WhiskConfig.dbProtocol)
+  val dbHostsList = WhiskProperties.getProperty(WhiskConfig.dbHostsList)
+  val dbPort = WhiskProperties.getProperty(WhiskConfig.dbPort)
+  val dbUsername = WhiskProperties.getProperty(WhiskConfig.dbUsername)
+  val dbPassword = WhiskProperties.getProperty(WhiskConfig.dbPassword)
+  val dbPrefix = WhiskProperties.getProperty(WhiskConfig.dbPrefix)
+  val dbWhiskAuth = WhiskProperties.getProperty(WhiskConfig.dbAuths)
+
+  private def getDockerCommand(host: String, component: String, cmd: String) = {
     def file(path: String) = Try(new File(path)).filter(_.exists).map(_.getAbsolutePath).toOption
+
     val docker = (file("/usr/bin/docker") orElse file("/usr/local/bin/docker")).getOrElse("docker")
 
-    val cmd = Seq(docker, "--host", host, "restart", component)
+    Seq(docker, "--host", host, cmd, component)
+  }
+
+  def restartComponent(host: String, component: String) = {
+    val cmd: Seq[String] = getDockerCommand(host, component, "restart")
+    println(s"Running command: ${cmd.mkString(" ")}")
+
+    TestUtils.runCmd(0, new File("."), cmd: _*)
+  }
+
+  def stopComponent(host: String, component: String) = {
+    val cmd: Seq[String] = getDockerCommand(host, component, "stop")
+    println(s"Running command: ${cmd.mkString(" ")}")
+
+    TestUtils.runCmd(0, new File("."), cmd: _*)
+  }
+
+
+  def startComponent(host: String, component: String) = {
+    val cmd: Seq[String] = getDockerCommand(host, component, "start")
     println(s"Running command: ${cmd.mkString(" ")}")
 
     TestUtils.runCmd(0, new File("."), cmd: _*)
   }
 
   def ping(host: String, port: Int) = {
-    val response = Try { Http().singleRequest(HttpRequest(uri = s"http://$host:$port/ping")).futureValue }.toOption
+    val response = Try {
+      Http().singleRequest(HttpRequest(uri = s"http://$host:$port/ping")).futureValue
+    }.toOption
+
+    response.map { res =>
+      (res.status, Unmarshal(res).to[String].futureValue)
+    }
+  }
+
+  def pingDB(host: String, port: Int) = {
```

Comment:
what do you think about using one ping method and passing the `path` as an argument.